app/controls/drag-drop.js

```diff
@@ -33,29 +30,35 @@ function notifyError (message) {
 }
 
 export default function dragDrop (event, files) {
-  if (!getIPFS()) {
+  const ipfs = getIPFS()
+  if (!ipfs) {
     notifyError('Can\'t upload file, IPFS Node is offline')
     return
   }
-  ipfsAPI.add(files, (err, res) => {
-    if (err || !res) {
-      logger.error(err)
-      notifyError(err || 'Failed to upload files')
-    }
 
-    logger.info('Uploading files', {files})
+  ipfs.add(files)
+    .then((res) => {
+      if (!res) {
+        notifyError('Failed to upload files')
+      }
+
+      logger.info('Uploading files', {files})
 
-    res.forEach((file) => {
-      const url = `https://ipfs.io/ipfs/${file.Hash}`
-      clipboard.writeText(url)
-      filesUploaded.push(file)
+      res.forEach((file) => {
+        const url = `https://ipfs.io/ipfs/${file.Hash}`
+        clipboard.writeText(url)
```

Comment:
I guess this works because in a context of adding a directory, the 'last file' is actually the root hash of the whole dir. This doesn't work as nicely when a group of files is added, they needed to be wrapped in a dir object (like `ipfs add path -w`)